common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -647,8 +945,175 @@ bool ap_wlan_hal_dwpal::sta_bss_steer(const std::string &mac, const std::string
+            LOG(ERROR) << "Autoconfiguration: unable to configure VAP " << vap_if << " SSID";
+            return false;
+        }
+
+        // Everything related to switching between open and WPA2
+        if (!update_vap_credentials_configure_wpa(vap_if, vap_hostapd_config, bss_info_conf)) {
+            // The function prints the error messages itself
+            return false;
+        }
+
+        // BSS type (backhaul, fronthaul or both)
+        // From hostap 2.8
+        // Enable Multi-AP functionality
+        // 0 = disabled (default)
+        // 1 = AP support backhaul BSS
+        // 2 = AP support fronthaul BSS
+        // 3 = AP supports both backhaul BSS and fronthaul BSS
+        int multi_ap = 0;
+        multi_ap |=
+            ((bss_info_conf.bss_type & WSC::eWscVendorExtSubelementBssType::BACKHAUL_BSS) != 0)
+                ? 0x01
+                : 0;
+        multi_ap |=
+            ((bss_info_conf.bss_type & WSC::eWscVendorExtSubelementBssType::FRONTHAUL_BSS) != 0)
+                ? 0x02
+                : 0;
+        if (!hostapd_config_set_value(vap_hostapd_config, "multi_ap", multi_ap)) {
+            LOG(ERROR) << "Autoconfiguration: cannot set " << vap_if << " multi-AP mode";
+            return false;
+        }
+
+        // Finally enable the VAP
+        if (!hostapd_config_set_value(vap_hostapd_config, "start_disabled", "0")) {
+            LOG(ERROR) << "Autoconfiguration: failed to enable VAP " << vap_if;
+        }
+
+        // Successfully updated the VAP, move on to the the next
+        ++hostapd_vap_iterator;
+    }
+
+    // If we are still here, then autoconfiguration was successful,
+    // update the hostapd configuration files and send RELOAD command.
+    if (!save_hostapd_config_file(fname, hostapd_config_head, hostapd_config_vaps)) {
+        LOG(ERROR) << "Autoconfiguration: cannot save hostapd config!";
+        return false;
+    }
+
+    // hostapd help says:
+    // RECONF [BSS name] = reconfigure interface (add/remove BSS's while other BSS are unaffected)
+    // if BSS name is given, that BSS will be reloaded (main BSS isn't supported)
+    // In reality (as of now, Jan 2020) only RECONF with explicitly specified interface name
+    // or sending SIGHUP work for making hostapd to reload the configuration.
+    size_t vap_ok_count = 0;
+    for (const auto &it : hostapd_config_vaps) {
+        std::string cmd("RECONF " + it.first);
+        if (!dwpal_send_cmd(cmd)) {
+            LOG(ERROR) << "Autoconfiguration: \"" << cmd << "\" command to hostapd has failed!";
+            // Keep going and try to complete what we can
+            continue;
+        }
+        ++vap_ok_count;
+    }
+
+    LOG(INFO) << "Autoconfiguration: completed successfully for " << vap_ok_count << " out of "
+              << hostapd_config_vaps.size() << " available hostapd BSS sections";
+    return (vap_ok_count == hostapd_config_vaps.size());
```

Comment:
We don't ned the OK count, just success, so I'd use `bool success = true;` instead. Also the final INFO message is redundant since we already report ERROR on each failure.

Reply:
I wanted the success message showing how many VAPs were processed (since that's the counter how many are available in the hostapd config to use by prplMesh), therefore the counter. This is a rare operation and the info message is quite useful here to confirm that auto-configuration has completed. I can change to debug if it really feels like it's polluting the logs.

Reply:
OK fair enough.